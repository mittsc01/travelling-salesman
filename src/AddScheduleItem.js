import React from 'react'

export default function AddScheduleItem(props){
    const routes = [
        {

                id: 1,
                name: 'Test'

        },
        {
                id: 2,
                name: 'Test2'
            },
       

    ]

    return (
    <form className="add-schedule">
        <select id="route" name="route" required>
    {routes.map((item,i) => <option key={`route-${i}`} value={item}>{item.name}</option>)}
        </select>
        <input type="date" required/>
        <button type='submit'>Submit</button>

    </form>
    )
}