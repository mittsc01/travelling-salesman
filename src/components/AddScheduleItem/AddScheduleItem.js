import React, { useEffect, useState } from 'react'
import RoutesService from '../../services/routes-service'


export default function AddScheduleItem(props) {
    const [routes, setRoutes] = useState([])
    const [schedule, setSchedule] = useState([])

    useEffect(() => {

        (async () => {
            const routes = await RoutesService.getRoutes(props)


            //routes.find(route => route.id === props.match.params.id)
            setRoutes(routes)
        })()
    }, [props])

    const handleSubmit = async (e) => {
        e.preventDefault()

        const postBody = {
            route_id: e.target.route.value,
            title: e.target.title.value,
            date: e.target.date.value
        }

        await RoutesService.postRun(postBody)
        props.history.push('/schedule')
    }

    return (
        <form className="add-schedule" onSubmit={handleSubmit}>
            <div className="form-div">
                <label htmlFor="route">Route</label>
                <select id="route" name="route" required>
                    {routes.length ? routes.map((item, i) => <option key={`route-${i}`} value={item.id}>{item.title}</option>) : null}
                </select>
                <label htmlFor="date">Date</label>
                <input type="date" name="date" id="date" required />
                <label htmlFor="title">Title</label>
                <input type="text" name="title" id="title" required />
            </div>

            <button className="save-button" type='submit'>Submit</button>

        </form>
    )
}