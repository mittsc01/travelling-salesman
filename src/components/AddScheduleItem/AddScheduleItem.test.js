import AddScheduleItem from './AddScheduleItem';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<AddScheduleItem />, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<AddScheduleItem />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });