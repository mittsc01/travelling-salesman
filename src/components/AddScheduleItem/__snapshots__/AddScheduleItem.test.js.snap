// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<form
  className="add-schedule"
  onSubmit={[Function]}
>
  <div
    className="form-div"
  >
    <label
      htmlFor="route"
    >
      Route
    </label>
    <select
      id="route"
      name="route"
      required={true}
    />
    <label
      htmlFor="date"
    >
      Date
    </label>
    <input
      id="date"
      name="date"
      required={true}
      type="date"
    />
    <label
      htmlFor="title"
    >
      Title
    </label>
    <input
      id="title"
      name="title"
      required={true}
      type="text"
    />
  </div>
  <button
    className="save-button"
    type="submit"
  >
    Submit
  </button>
</form>
`;
